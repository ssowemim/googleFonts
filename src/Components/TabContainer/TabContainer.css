.TabContainer {
  padding: 15px 20px 5px; }
  .TabContainer .Text__bold {
    font-weight: bold; }
  .TabContainer .TabContainer__Tab--Content {
    font-size: 14px;
    padding-top: 15px;
    padding-bottom: 10px; }
    .TabContainer .TabContainer__Tab--Content br {
      line-height: 1.3; }
    .TabContainer .TabContainer__Tab--Content h4 {
      font-size: 16px;
      margin-bottom: 10px; }
    .TabContainer .TabContainer__Tab--Content .Info-Font {
      font-family: inherit;
      background: rgba(226, 225, 215, 0.6);
      padding: 15px 10px;
      margin-bottom: 0;
      margin-top: 10px;
      display: flex;
      justify-content: space-between;
      align-items: center; }
      .TabContainer .TabContainer__Tab--Content .Info-Font .fa {
        margin-left: 10px;
        cursor: pointer; }
    .TabContainer .TabContainer__Tab--Content .Link {
      margin-top: 20px;
      display: inline-block;
      margin-left: 10px;
      color: #000; }
      .TabContainer .TabContainer__Tab--Content .Link:hover {
        text-decoration: none; }
  .TabContainer .react-tabs__tab-list {
    display: flex; }
    .TabContainer .react-tabs__tab-list .react-tabs__tab {
      margin-right: 10px;
      text-transform: uppercase;
      cursor: pointer;
      padding: 6px 10px; }
      .TabContainer .react-tabs__tab-list .react-tabs__tab:hover, .TabContainer .react-tabs__tab-list .react-tabs__tab.react-tabs__tab--selected {
        border-bottom: 2px solid #ffdf01; }
