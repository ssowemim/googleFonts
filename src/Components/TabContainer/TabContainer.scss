.TabContainer {
  padding: 15px 20px 5px;

  .Text__bold {
    font-weight: bold;
  }

  .TabContainer__Tab--Content {
    font-size: 14px;
    padding-top: 15px;
    padding-bottom: 10px;

    br {
      line-height: 1.3;
    }

    h4 {
      font-size: 16px;
      margin-bottom: 10px;
    }

    .Info-Font {
      font-family: inherit;
      background: rgba(226, 225, 215, 0.6);
      padding: 15px 10px;
      margin-bottom: 0;
      margin-top: 10px;
      display: flex;
      justify-content: space-between;
      align-items: center;

      .fa {
        margin-left: 10px;
        cursor: pointer;
      }
    }

    .Link {
      margin-top: 20px;
      display: inline-block;
      margin-left: 10px;
      color: #000;

      &:hover {
        text-decoration: none;
      }
    }

  }
  
  .react-tabs__tab-list {
    display: flex;
    .react-tabs__tab {
      margin-right: 10px;
      text-transform: uppercase;
      cursor: pointer;
      padding: 6px 10px;

      &:hover, &.react-tabs__tab--selected {
        border-bottom: 2px solid #ffdf01;
      }
    }
  }
}