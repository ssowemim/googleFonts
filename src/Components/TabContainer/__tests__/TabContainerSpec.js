import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';
import TabContainer from './../index';

describe('TabContainer', () => {
    it('renders correctly', () => {
        let tree = renderer.create(
            <TabContainer fontName="oswald" />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly font-family has space', () => {
        let tree = renderer.create(
            <TabContainer fontName="some font family" />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
