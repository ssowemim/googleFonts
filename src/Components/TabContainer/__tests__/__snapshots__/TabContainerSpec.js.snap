// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabContainer renders correctly 1`] = `
<div
  className="TabContainer"
>
  <div
    className="react-tabs"
    data-tabs={true}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <ul
      className="react-tabs__tab-list"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-1"
        aria-disabled="false"
        aria-selected="true"
        className="react-tabs__tab react-tabs__tab--selected"
        id="react-tabs-0"
        role="tab"
        tabIndex="0"
      >
        Standard
      </li>
      <li
        aria-controls="react-tabs-3"
        aria-disabled="false"
        aria-selected="false"
        className="react-tabs__tab"
        id="react-tabs-2"
        role="tab"
        tabIndex={null}
      >
        @Import
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-0"
      className="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-1"
      role="tabpanel"
    >
      <div
        className="TabContainer__Tab--Content"
      >
        <h4>
          Embed Font
        </h4>
        To embed your selected fonts into a webpage, copy this code into the &lt;head&gt; of your HTML document.
        <div
          className="Info-Font"
        >
          <span>
            &lt;link href="https://fonts.googleapis.com/css?family=
            <span
              className="Text__bold"
            >
              oswald
            </span>
            " rel="stylesheet"&gt;
          </span>
          <div
            className="CopyContent"
          >
            <i
              className="fa fa-copy"
              data-item="<link href=\\"https://fonts.googleapis.com/css?family=oswald\\" rel=\\"stylesheet\\">"
              onClick={[Function]}
              onMouseLeave={[Function]}
            />
            <span
              className="CopyContent__TitlePopUp"
            >
              Click to Copy
            </span>
          </div>
        </div>
      </div>
      <div
        className="TabContainer__Tab--Content"
      >
        <h4>
          Specify in CSS
        </h4>
        Use the following CSS rules to specify these families:
        <div
          className="Info-Font"
        >
          <span>
            font-family: 
            oswald, sans-serif
            ;
          </span>
          <div
            className="CopyContent"
          >
            <i
              className="fa fa-copy"
              data-item="font-family: oswald, sans-serif;"
              onClick={[Function]}
              onMouseLeave={[Function]}
            />
            <span
              className="CopyContent__TitlePopUp"
            >
              Click to Copy
            </span>
          </div>
        </div>
        <strong>
          Goolgle Fonts:
        </strong>
        <a
          className="Link"
          href="https://fonts.google.com/specimen/oswald"
          target="_blank"
        >
          https://fonts.google.com/specimen/
          oswald
        </a>
      </div>
    </div>
    <div
      aria-labelledby="react-tabs-2"
      className="react-tabs__tab-panel"
      id="react-tabs-3"
      role="tabpanel"
    />
  </div>
</div>
`;

exports[`TabContainer renders correctly font-family has space 1`] = `
<div
  className="TabContainer"
>
  <div
    className="react-tabs"
    data-tabs={true}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <ul
      className="react-tabs__tab-list"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-5"
        aria-disabled="false"
        aria-selected="true"
        className="react-tabs__tab react-tabs__tab--selected"
        id="react-tabs-4"
        role="tab"
        tabIndex="0"
      >
        Standard
      </li>
      <li
        aria-controls="react-tabs-7"
        aria-disabled="false"
        aria-selected="false"
        className="react-tabs__tab"
        id="react-tabs-6"
        role="tab"
        tabIndex={null}
      >
        @Import
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-4"
      className="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-5"
      role="tabpanel"
    >
      <div
        className="TabContainer__Tab--Content"
      >
        <h4>
          Embed Font
        </h4>
        To embed your selected fonts into a webpage, copy this code into the &lt;head&gt; of your HTML document.
        <div
          className="Info-Font"
        >
          <span>
            &lt;link href="https://fonts.googleapis.com/css?family=
            <span
              className="Text__bold"
            >
              some+font+family
            </span>
            " rel="stylesheet"&gt;
          </span>
          <div
            className="CopyContent"
          >
            <i
              className="fa fa-copy"
              data-item="<link href=\\"https://fonts.googleapis.com/css?family=some font family\\" rel=\\"stylesheet\\">"
              onClick={[Function]}
              onMouseLeave={[Function]}
            />
            <span
              className="CopyContent__TitlePopUp"
            >
              Click to Copy
            </span>
          </div>
        </div>
      </div>
      <div
        className="TabContainer__Tab--Content"
      >
        <h4>
          Specify in CSS
        </h4>
        Use the following CSS rules to specify these families:
        <div
          className="Info-Font"
        >
          <span>
            font-family: 
            some font family, sans-serif
            ;
          </span>
          <div
            className="CopyContent"
          >
            <i
              className="fa fa-copy"
              data-item="font-family: some font family, sans-serif;"
              onClick={[Function]}
              onMouseLeave={[Function]}
            />
            <span
              className="CopyContent__TitlePopUp"
            >
              Click to Copy
            </span>
          </div>
        </div>
        <strong>
          Goolgle Fonts:
        </strong>
        <a
          className="Link"
          href="https://fonts.google.com/specimen/some+font+family"
          target="_blank"
        >
          https://fonts.google.com/specimen/
          some+font+family
        </a>
      </div>
    </div>
    <div
      aria-labelledby="react-tabs-6"
      className="react-tabs__tab-panel"
      id="react-tabs-7"
      role="tabpanel"
    />
  </div>
</div>
`;
